v2/cli/init_command.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+
+	"github.com/AlecAivazis/survey/v2"
+	"github.com/urfave/cli/v2"
+)
+
+type InitFields struct {
+	ID       string
+	Language string
+	Git      string
+}
+
+var initCommand = &cli.Command{
+	Name:  "init",
+	Usage: "Initialize a template Brigade project",
+	Flags: []cli.Flag{
+		&cli.StringFlag{
+			Name:    flagID,
+			Aliases: []string{"i"},
+			Usage: "Name of the Brigade project to create " +
+				"(required)",
+			Required: true,
+		},
+		&cli.StringFlag{
+			Name:    flagLanguage,
+			Aliases: []string{"l"},
+			Usage:   "Set the language of the brigade project file to either JavaScript or TypeScript (default)",
+		},
+		&cli.StringFlag{
+			Name:    flagGit,
+			Aliases: []string{"g"},
+			Usage:   "Initialize the brigade project from an existing git repo ",
+		},
+	},
+	Action: initialize,
+}
+
+// Template project.yaml file for a non-git project
+var newProjectFileBytes = []byte(
+	`# yaml-language-server: $schema=https://raw.githubusercontent.com/brigadecore/brigade/v2/v2/apiserver/schemas/project.json
+apiVersion: brigade.sh/v2-alpha.5
+kind: Project
+metadata:
+	id: {{ .ID }}
+description: Your newly-initialized Brigade project!
+spec:
+	workerTemplate:
+		logLevel: DEBUG
+		defaultConfigFiles:
+			brigade.{{ .Language }}: |
+				console.log("Hello, World!");
+`)
+
+// Template project.yaml file for a user who provided a git repo link
+var newGitProjectFileBytes = []byte(
+	`# yaml-language-server: $schema=https://raw.githubusercontent.com/brigadecore/brigade/v2/v2/apiserver/schemas/project.json
+apiVersion: brigade.sh/v2-alpha.5
+kind: Project
+metadata:
+	id: {{ .ID }}
+description: Your newly-initialized Brigade project!
+spec:
+	workerTemplate:
+		logLevel: DEBUG
+		git:
+			cloneURL: {{ .Git }}
+`)
```

Comment:
Let's try to consolidate these into a single template.

I say this because right now you have one template for a project that uses git and another for a project that doesn't. Suppose, however, I told you that the default script need to be a little more complex than just `console.log("Hello, World!")`. Now you could be in a position of having a template with git support + a javascript (with JS style imports and such) and a template with git support + a typescript (with TS style imports and such). The number of permutations grows exponentially with the number of options we expose, so it's better to have one template that has some conditional logic in it.

There is what I might describe as an "expression language" called [sprig](https://github.com/Masterminds/sprig) that you can use in order to do conditionals and other complex things within your templates.